js/samples/quickstart-java/src/main/webapp/index.jsp

```diff
@@ -1,95 +1,106 @@
+                    washing. Throughout history, people have settled near fresh water. Living near a water source helps
+                    ensure that people have the water they need. There was an added bonus, too. Water could be used as a
+                    travel route for people and goods. Many Americans live near popular water sources, such as the
+                    Mississippi River, the Colorado River and the Great Lakes.</p>
+                <p>Mountains and deserts have been settled by fewer people than the plains areas. However, they have
+                    valuable resources of their own.</p>
             </div>
-            <div class="navbar-collapse collapse">
-                <ul class="nav navbar-nav">
-                    <li><a href="/">Home</a></li>
-                </ul>
-            </div>
-        </div>
-    </nav>
-    <div class="container body-content">
+        </article>
+    </main>
 
-    <meta name='viewport' content='width=device-width, initial-scale=1'>
+    <div id='message'></div>
+
+    <script type='text/javascript'>
+        function handleLaunchImmersiveReader() {
+            $('#message').hide();
 
-    <h1 id='title'>Geography</h1>
-    <span id='content'>
-        <p>The study of Earth's landforms is called physical geography. Landforms can be mountains and valleys. They can also be glaciers, lakes or rivers. Landforms are sometimes called physical features. It is important for students to know about the physical geography of Earth. The seasons, the atmosphere and all the natural processes of Earth affect where people are able to live. Geography is one of a combination of factors that people use to decide where they want to live.</p>
-        <p>The physical features of a region are often rich in resources. Within a nation, mountain ranges become natural borders for settlement areas. In the U.S., major mountain ranges are the Sierra Nevada, the Rocky Mountains, and the Appalachians.</p>
-        <p>Fresh water sources also influence where people settle. People need water to drink. They also need it for washing. Throughout history, people have settled near fresh water. Living near a water source helps ensure that people have the water they need. There was an added bonus, too. Water could be used as a travel route for people and goods. Many Americans live near popular water sources, such as the Mississippi River, the Colorado River and the Great Lakes.</p>
-        <p>Mountains and deserts have been settled by fewer people than the plains areas. However, they have valuable resources of their own.</p>
-    </span>
+            getTokenAndSubdomainAsync().then(function (response) {
+                const token = response['token'];
+                const subdomain = response['subdomain'];
 
-    <button onclick='handleLaunchImmersiveReader()'>
-        Immersive Reader
-    </button>
+                const data = {
+                    title: $('#ir-title').text().trim(),
```

Comment:
I think `text()` might preserve the new lines from the HTML file. Normally we don't want to do this because the browser ignores that. Can you double check the behavior here?
If it is preserving the new lines, changing it to the original `document.getElementById('...').innerText` might be a better option.

Reply:
I'll add this as a followup to do on all samples, in another PR